### PR TITLE
Add the framework for a chrome app.

### PR DESCRIPTION
**Pull request by chuckhays**
* Adds the required files for a chrome app
* Generates a random room id and connects

CC: @juberti @jiayliu

### CODE REVIEW

File: samples/web/content/apprtc/js/appwindow.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+/* More information about these options at jshint.com/docs/options */
+// Variables defined in and used from main.js.
+/* globals randomString, initialize */
+/* exported params */
+'use strict';
+
+// Provide params var, provided inline in index.html.
+var params;
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/appwindow.js]: duplicated declaration of params

File: samples/web/content/apprtc/js/appwindow.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+// Generate random room id and connect.
+var room = randomString(9);
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/appwindow.js]: Seems unnecessary to save it in a new var, e.g. you can do 
params.roomId = randomString(9)

File: samples/web/content/apprtc/js/appwindow.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  videoSendCodec: '',
+  videoRecvBitrate: '',
+  videoRecvCodec: '',
+  wssUrl: 'wss://apprtc-ws.webrtc.org:443/ws',
+  wssPostUrl: 'https://apprtc-ws.webrtc.org:443'
+};
+
+params.roomId = room;
+params.roomLink =  'https://apprtc.appspot.com/room/' + room;
+params.server = 'https://apprtc.appspot.com';
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/appwindow.js]: Suggest to rename to roomServer to match other clients.

File: samples/web/content/apprtc/js/main.js [hunk 4]
```diff
@@ -139,7 +139,12 @@ function disconnectFromRoom() {
   // When the other client sees BYE it attempts to post offer and candidates to
   // GAE. GAE needs to know that we're disconnected at that point otherwise
   // it will forward messages to this client instead of storing them.
-  path = '/bye/' + params.roomId + '/' + params.clientId;
+  var server = params.server;
+  if (!server) {
+    // If server is not provided, use relative URI.
+    server = '';
+  }
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/main.js]: This is duplicated in a few places. Why not set params.server at the beginning of initialize()?

File: samples/web/content/apprtc/js/utils_test.js [hunk 5]
```diff
@@ -43,3 +43,22 @@ UtilsTest.prototype.testFilterTurnUrlsTcp = function() {
...
+UtilsTest.prototype.testRandomReturnsCorrectLength = function() {
+  assertEquals('13 length string', 13, randomString(13).length);
+  assertEquals('5 length string', 5, randomString(5).length);
+  assertEquals('10 length string', 10, randomString(10).length);
+};
+
+UtilsTest.prototype.testRandomReturnsCorrectCharacters = function() { 
+  var str = randomString(500);
+
+  // randromString should return only the digits 0-9
```
Comment (jiayliu) [inline samples/web/content/apprtc/js/utils_test.js]: nit: end the comment with '.'

File: samples/web/content/apprtc/appwindow.html [hunk 6]
```diff
@@ -0,0 +1,51 @@
+<!DOCTYPE html>
```
Comment (jiayliu) [inline samples/web/content/apprtc/appwindow.html]: Can we have a script to generate this from index.html?

File: samples/web/content/apprtc/js/appwindow.js (hunk 1 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/appwindow.js]: Fixed.

File: samples/web/content/apprtc/js/appwindow.js (hunk 2 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/appwindow.js]: Done.

File: samples/web/content/apprtc/js/appwindow.js (hunk 3 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/appwindow.js]: Done.

File: samples/web/content/apprtc/js/main.js (hunk 4 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/main.js]: Good idea, done.

File: samples/web/content/apprtc/js/utils_test.js (hunk 5 above)
Comment (chuckhays) [inline samples/web/content/apprtc/js/utils_test.js]: Done. I saw this right after I sent the PR.

File: samples/web/content/apprtc/appwindow.html (hunk 6 above)
Comment (chuckhays) [inline samples/web/content/apprtc/appwindow.html]: Yes, planning to do this, if it is useful, when we create the build script that packages the app. It isn't clear to me yet how similar index.html and appwindow.html wll be after we add all the new UI we are currently discussing.

Comment (jiayliu) [inline samples/web/content/apprtc/appwindow.html]: I think the UI for web and Chrome OS will be identical for the most part,
except:
1. the registration view, which should be shown/hidden dynamically anyway.
2. the contacts view, which is unclear how it works on Chrome OS.

So I think a script to avoid these two UI out of sync should be useful for
a long time.

On Wed, Jan 7, 2015 at 3:39 PM, Chuck Hays <notifications@github.com> wrote:

> In samples/web/content/apprtc/appwindow.html
> <https://github.com/GoogleChrome/webrtc/pull/349#discussion-diff-22625816>
> :
>
> > @@ -0,0 +1,51 @@
> > +<!DOCTYPE html>
>
> Yes, planning to do this, if it is useful, when we create the build script
> that packages the app. It isn't clear to me yet how similar index.html and
> appwindow.html wll be after we add all the new UI we are currently
> discussing.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/webrtc/pull/349/files#r22625816>.
>



-- 
Jiayang Liu

Comment (chuckhays) [inline samples/web/content/apprtc/appwindow.html]: I think this will be the next thing that I do (the build/packaging steps), in a separate CL.

Comment (chuckhays) [inline samples/web/content/apprtc/appwindow.html]: I did this work in another branch:
https://github.com/chuckhays/webrtc/commit/e1be7f0e016453a8585319eef7e601d3587595cb
Do you want me to merge it to this branch, or do it as a separate PR after this one is completed?

Comment (jiayliu) [inline samples/web/content/apprtc/appwindow.html]: Thanks! A separate CL sounds fine to me.

**Comment by jiayliu** [comment]
lgtm
